### PR TITLE
enhance accuracy for model kimi-vl-instruct-a3b

### PR DESCRIPTION
**Pull request by McZyWu**
Co-authored-by:  randgun@github.com hexq0210@github.com 
## Motivation
previously accuracy for kimi-vl-instruct-a3b is no more than 0.05.

## Modifications
as follows

## Accuracy Tests
eagle mode: accuracy is at least 0.675.
graph mode: accuracy is at least 0.67

### CODE REVIEW

File: python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def forward_mha_prepare_npu(
         k_pe = latent_cache[:, :, m.kv_lora_rank:]
         if m.rotary_emb is not None:
             q_pe, k_pe = m.rotary_emb(positions, q_pe, k_pe)
+        m._set_mla_kv_buffer(latent_cache, kv_a, k_pe, forward_batch)
```
Comment (randgun) [inline python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py]: Do not use callback func, use `forward_batch.token_to_kv_pool.set_kv_buffer(
            m, forward_batch.out_cache_loc, kv_a.unsqueeze(1), k_pe
        )` instead and delete npu part in _set_mla_kv_buffer.

File: python/sglang/srt/hardware_backend/npu/moe/topk.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def fused_topk_npu(
             )
         topk_weights = topk_weights.to(torch.float32)
 
-    elif use_grouped_topk and correction_bias is not None:
+    elif use_grouped_topk and correction_bias is not None and router_logits.shape[-1] == 256:
```
Comment (randgun) [inline python/sglang/srt/hardware_backend/npu/moe/topk.py]: npu_moe_gating_top_k has supported router_logits.shape[-1] <= 2048 in pta 7.2.0 version.

File: python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py (hunk 1 above)
Comment (McZyWu) [reply python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py]: Thank you. Your reviews are extremely valuable. I've adopted.

File: python/sglang/srt/hardware_backend/npu/moe/topk.py (hunk 2 above)
Comment (McZyWu) [reply python/sglang/srt/hardware_backend/npu/moe/topk.py]: Thank you. Your advice has just helped me make the changes. Adopted
